executor/insert_common.go

```diff
@@ -1051,21 +1051,26 @@ func (e *InsertValues) batchCheckAndInsert(ctx context.Context, rows [][]types.D
 	}
 	prefetchStart := time.Now()
 	// Fill cache using BatchGet, the following Get requests don't need to visit TiKV.
-	if _, err = prefetchUniqueIndices(ctx, txn, toBeCheckedRows); err != nil {
-		return err
+	// Temporary table need not to do prefetch because its all data are stored in the memory.
+	if e.Table.Meta().TempTableType == model.TempTableNone {
+		if _, err = prefetchUniqueIndices(ctx, txn, toBeCheckedRows); err != nil {
```

Comment:
Could it be moved into `perfetchUniqueIndices` just like here?
https://github.com/pingcap/tidb/pull/26636/files#diff-6024133129a4038bfa75513ee2ac6a664c9c184fbf026997ec69f6db006aa38cR178

Reply:
The question is what is the return of `perfetchUniqueIndices`. If it returns an empty data, it seems not fit the semantics of the method. If it returns the data in temporary table, it seems unnecessary and no one depends the return to do anything...